### ISSUE TITLE
Creating admin api tokens possible?

### ISSUE THREAD

**Issue by stephanrenggli** [CTFd/CTFd#1891]
**Environment**:

- CTFd Version/Commit: 3.3.0
- Operating System: docker/linux
- Web Browser and Version: chrome 90.0

**What happened?**

I would like to create an admin API token, else the API cannot see hidden teams/users/challenges etc.
Currently when im logged in as an admin user and go to ctfdurl/settings#tokens and create a token it says "type: user".

Is this the intended behaviour?
If the API user does not have admin permissions, I can create hidden teams/users/challenges but I cannot list them via the API bulk endpoint. I can access data on hidden items via API using their ID.
But if I create the team via API and don't store the ID, I won't be able to see it again unless I use the GUI.

**What did you expect to happen?**

Being able to create an API token with admin permissions.

**How to reproduce your issue**

Go to ctfdurl/settings#tokens and create a token.

**Any associated stack traces or error logs**

None

**Comment by ColdHeat** [comment]
API tokens that belong to an admin user have admin privileges. The user type that's specified there is just to distinguish it from any other type of token. We should probably hide that field entirely actually.

**Comment by stephanrenggli** [comment]
Thank you for the response.

So my token does have admin access but I cannot see hidden items using the API.
Shouldn't the admin user be able to see the same things via GUI and via API. Currently the two methods will provide different results.
GUI = I can see hidden items
API = I cannot see hidden items (unless I access them by ID)

**Comment by stephanrenggli** [comment]
I tried to put everything important in the screenshot below, I hope this helps:

![ctfd](https://user-images.githubusercontent.com/5135902/119036635-3be2c780-b9b1-11eb-93fb-23eb5d62b4ab.png)

Left side: local CTFd instance with 3 users:
- ctfd (admin) <- this user created my api token
- visibleuser (visible)
- hiddenuser (hidden)

Top right:
- API request to /api/v1/users <- supposed to list all users?
- Authorization header set to "Token <api_token>"
- Content-Type header set to "application/json"

-> note this does only return 1 user "visibleuser" which is the only visible user registered in CTFd

Bottom right:
- API request to /api/v1/users/1 <- supposed to return the user with user ID 1 (user ctfd (hidden) in my case
- Authorization header set to "Token <api_token>"
- Content-Type header set to "application/json"

-> like this I can get information on hidden users (ctfd) in this case

From what I can see the API requests are executed correctly, as they do return valid results. But hidden users are not returned via the bulk endpoint. Maybe I am missing something obvious?

Appreciate your help!

**Comment by ColdHeat** [comment]
Ah sorry I missed what you were looking for. Some endpoints like that since they're shared with users and admins need a a flag to be viewed as admin. 

Try using `/api/v1/users?view=admin`. Here's the relevant code: https://github.com/CTFd/CTFd/blob/f00e69d6193a746d30092e60f9a0c7a41b6f4fa3/CTFd/api/v1/users.py#L102-L107

**Comment by stephanrenggli** [comment]
Thank you!
Works as expected now.

From what I can tell this isn't documented in the swagger UI. Maybe this could be added in the future?

**Comment by ColdHeat** [comment]
Yeah it should be added. It's just been a little difficult to work with Swagger. I have perhaps some ideas about migrating to a better API codebase like FastAPI but it will take time.
